Statsman/PeriodStatisticsService.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Statsman.Core.TimeSplitting;
+using Statsman.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Common.Models;
+using Watchman.Cqrs;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Messages.Queries;
+using Watchman.Integrations.Quickchart;
+
+namespace Statsman
+{
+    public class PeriodStatisticsService
+    {
+        private readonly IQueryBus queryBus;
+        private readonly TimeSplittingService timeSplittingService = new TimeSplittingService(); //TODO IoC
+        private readonly ChartsService _chartsService = new ChartsService();
+
+        public PeriodStatisticsService(IQueryBus queryBus)
+        {
+            this.queryBus = queryBus;
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerMinute(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.UtcNow.AddMinutes(-request.TimeBehind.TotalMinutes), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDetailedPeriod(request, timeRange, Period.Minute);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per minute"),
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Minute, timeRange)); //TODO get label and message from configuration
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerHour(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.UtcNow.AddHours(-request.TimeBehind.TotalHours), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDetailedPeriod(request, timeRange, Period.Hour);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per hour"),
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Hour, timeRange));
+        }
+
```

Comment:
w tym miejscu można wydzielić do innej metody wspólny kod. W zasadzie tutaj dzieje się to samo tylko na innych danych.